src/Components/RegisterForm.js

```diff
@@ -44,24 +81,29 @@ class RegisterForm extends Component {
         <header className="user__header">
             <h1 className="user__title">Sign-up</h1>
         </header>
-        <form className="form" onSubmit = {this.createUser}>
+        <form className="form" onSubmit={this.createUser}>
             <div className="form__group">
-                <input type="text"  placeholder="Username" className="form__input" onChange = {(event) => this.handleInputChange(event, 'userName')}/>
+                <input type="text"  placeholder="Username" className="form__input" onChange={(event) => this.handleInputChange(event, 'userName')}/>
+                <p>{this.state.userName.error}</p>
             </div>
 
             <div className="form__group">
-                <input type="email"  placeholder="Email" className="form__input" onChange = {(event) => this.handleInputChange(event, 'email')}/>
+                <input type="email"  placeholder="Email" className="form__input" onChange={(event) => this.handleInputChange(event, 'email')}/>
+                <p>{this.state.email.error}</p>
             </div>
 
             <div className="form__group">
-                <input type="password"  name="password" placeholder="Password" className="form__input" onChange = {(event) => this.handleInputChange(event, 'password')}/>
+                <input type="password"  name="password" placeholder="Password" className="form__input" onChange={(event) => this.handleInputChange(event, 'password')}/>
+                <p>{this.state.password.error}</p>
             </div>
 
             <div className="form__group">
-                <input type="password"  name="cPassword" placeholder="Confirm password" className="form__input" onChange = {(event) => this.handleInputChange(event, 'cPassword')}/>
+                <input type="password"  name="cPassword" placeholder="Confirm password" className="form__input" onChange={(event) => this.handleInputChange(event, 'cPassword')}/>
+                <p>{this.state.cPassword.error}</p>
             </div>
             <button className="btn"  type="submit">Register</button>
-            {this.props.registerSuccess && <Redirect to='/Login'/>}
+            <p>{this.props.error}</p>
+            { this.props.registerSuccess &&  <Redirect to="/Login"/> }
```

Comment:
Put it to the top of render.
You don't need to render the form if register succesed, you just need to redirect